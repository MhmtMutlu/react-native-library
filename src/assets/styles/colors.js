const colors = {
  grey: '#6c757d',
  darkGrey: '#495057',
  darkGrey2x: '#343a40',
  lightGrey: '#adb5bd',
  lightGrey2x: '#dee2e6',
  lightGrey3x: '#e0e0e0',
  white: '#f8f9fa',
};

export default colors;
