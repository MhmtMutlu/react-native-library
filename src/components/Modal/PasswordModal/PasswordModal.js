import React from 'react';
import {Text, View} from 'react-native';
import Modal from 'react-native-modal';

import styles from './PasswordModal.styles';

const PasswordModal = ({isVisible, onClose}) => {
  const modalItem = (text, number) => (
    <Text style={styles.text}>
      <Text style={styles.text_title}>{`${number}-`}</Text> {text}
    </Text>
  );

  return (
    <Modal
      testID="test-password-modal"
      style={styles.modal}
      isVisible={isVisible}
      swipeDirection="down"
      onSwipeComplete={onClose}
      onBackdropPress={onClose}
      onBackButtonPress={onClose}>
      <View testID="test-password-modal-container" style={styles.container}>
        {modalItem('Şifreniz en az 1 adet küçük karakter içermelidir!', 1)}
        {modalItem('Şifreniz en az 1 adet büyük karakter içermelidir!', 2)}
        {modalItem('Şifreniz en az 1 adet sayı içermelidir!', 3)}
        {modalItem('Şifreniz en az 1 adet özel karakter içermelidir!', 4)}
        {modalItem('Şifreniz en az 8 karakter olmalıdır!', 5)}
      </View>
    </Modal>
  );
};

export default PasswordModal;
