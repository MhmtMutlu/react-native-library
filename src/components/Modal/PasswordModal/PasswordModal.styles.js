import {StyleSheet, Dimensions} from 'react-native';
import colors from '../../../assets/styles/colors';

const deviceSize = Dimensions.get('window');

export default StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    padding: 20,
    borderRadius: 15,
    height: deviceSize.height / 3,
  },
  modal: {
    left: deviceSize.width / 100,
    right: deviceSize.width / 100,
    position: 'absolute',
    top: deviceSize.height / 3.5,
    marginVertical: 20,
  },
  text: {
    fontSize: 18,
    marginBottom: 10,
    fontStyle: 'italic',
  },
  text_title: {
    fontWeight: 'bold',
    fontSize: 18,
  },
});
