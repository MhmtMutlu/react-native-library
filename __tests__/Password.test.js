import React from 'react';
import {render} from '@testing-library/react-native';
import PasswordModal from '../src/components/Modal/PasswordModal';
import styles from '../src/components/Modal/PasswordModal/PasswordModal.styles';

test('should match with snapshot', () => {
  const comp = render(<PasswordModal isVisible={true} />);
  expect(comp).toMatchSnapshot();
});

test('check visible prop', () => {
  const comp = render(<PasswordModal isVisible={true} />);
  const contentModal = comp.getByTestId('test-password-modal').props;
  expect(contentModal).toMatchObject({
    visible: true,
  });
});

test('should render PasswordModal css', () => {
  const comp = render(<PasswordModal isVisible={true} />);
  const passwordModal = comp.getByTestId('test-password-modal-container').props
    .style;
  expect(passwordModal).toMatchSnapshot(styles.container);
});
