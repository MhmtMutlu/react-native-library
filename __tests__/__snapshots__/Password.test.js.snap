// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection="down"
  swipeThreshold={100}
  testID="test-password-modal"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "left": 7.5,
        "margin": 37.5,
        "marginVertical": 20,
        "position": "absolute",
        "right": 7.5,
        "top": 381.14285714285717,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#f8f9fa",
          "borderRadius": 15,
          "height": 444.6666666666667,
          "padding": 20,
        }
      }
      testID="test-password-modal-container"
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontStyle": "italic",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          1-
        </Text>
         
        Şifreniz en az 1 adet küçük karakter içermelidir!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontStyle": "italic",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          2-
        </Text>
         
        Şifreniz en az 1 adet büyük karakter içermelidir!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontStyle": "italic",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          3-
        </Text>
         
        Şifreniz en az 1 adet sayı içermelidir!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontStyle": "italic",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          4-
        </Text>
         
        Şifreniz en az 1 adet özel karakter içermelidir!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontStyle": "italic",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          5-
        </Text>
         
        Şifreniz en az 8 karakter olmalıdır!
      </Text>
    </View>
  </View>
</Modal>
`;

exports[`should render PasswordModal css 1`] = `
Object {
  "backgroundColor": "#f8f9fa",
  "borderRadius": 15,
  "height": 444.6666666666667,
  "padding": 20,
}
`;
